src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchive.cs

```diff
@@ -211,6 +212,20 @@ public ZipArchive(Stream stream, ZipArchiveMode mode, bool leaveOpen, Encoding?
             }
         }
 
+        /// <summary>
+        /// Gets or sets the optional archive comment.
+        /// </summary>
+        /// <remarks>
+        /// The comment encoding is determined by the <c>entryNameEncoding</c> parameter of the <see cref="ZipArchive(Stream,ZipArchiveMode,bool,Encoding?)"/> constructor.
+        /// If the comment byte length is larger than <see cref="ushort.MaxValue"/>, it will be truncated when disposing the archive.
+        /// </remarks>
+        [AllowNull]
+        public string Comment
+        {
+            get => ZipHelper.GetTruncatedComment(_archiveComment, EntryNameAndCommentEncoding, ZipEndOfCentralDirectoryBlock.ZipFileCommentMaxLength);
```

Comment:
Few questions:

1. What is the point of converting a string into bytes and into string again? Why don't you just return `_archiveComment` here?
2. What is the advantage of converting byte array into string in the `ReadEndOfCentralDirectory` method? With the previous implementation (that was storing the byte array) the users would not need to pay for the conversion. Now no matter if they do want to access the `string Comment`, a byte array is converted into a string. And once they do want to get the string, it's again converted into byte array and into string again.

Reply:
> With the previous implementation (that was storing the byte array) the users would not need to pay for the conversion.

There's no "previous implementation". Before, the user was unable to set a comment. Now, the user can set it, change it, and retrieve it, so we need to adapt to all that.

> Now no matter if they do want to access the string Comment, a byte array is converted into a string. And once they do want to get the string, it's again converted into byte array and into string again.

I think we should get in a call to discuss the changes in this PR, because I already went back and forth with the implementation 3 times due to PR suggestions requesting opposite behaviors. My code was initially doing what you are mentioning: storing the converted bytes, storing the string. Then I was asked to simplify the setter, and then I was asked to only update the bytes in the getter. Now it seems the request is to cache everything again.

> What is the point of converting a string into bytes and into string again? Why don't you just return _archiveComment here?

The spec specifies that the comment cannot be larger than a predefined size, in bytes (`uint.MaxSize`).

If the user saves a comment larger than that length, we need to truncate it.

If the user sets a comment that's too long, then calls the getter, we should be clear that we truncated it, so we need to convert it to bytes, truncate the bytes (aligning the truncation to the char size), then convert the truncated bytes back to string, and return that. We need to do the same when saving to the archive or the entry.

It would be confusing for the user if we returned the whole comment, but then they will realize that we only stored part of it in the archive or the entry.

Reply:
@carlossanlop this is decoding-encoding-decoding just to truncate the comment and present it as `string`.

you can avoid the double decoding by only maintaining a `byte[]` field (instead of `string _archiveComment`), decode it in the getter and save the string as encoded in the setter.

Reply:
Ok, pushed the proposed change. PTAL.